pkg/splunk/enterprise/afwscheduler_test.go

```diff
@@ -833,7 +847,506 @@ func TestAfwGetReleventStatefulsetByKind(t *testing.T) {
+	worker.appDeployInfo.ObjectHash = ""
+
+	client := spltest.NewMockClient()
+
+	// Create S3 secret
+	s3Secret := spltest.GetMockS3SecretKeys("s3-secret")
+
+	client.AddObject(&s3Secret)
+
+	// Create namespace scoped secret
+	_, err := splutil.ApplyNamespaceScopedSecretObject(client, "test")
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+
+	splclient.RegisterS3Client("aws")
+	// Update the GetS3Client with our mock call which initializes mock AWS client
+	getClientWrapper := splclient.S3Clients["aws"]
+	getClientWrapper.SetS3ClientFuncPtr("aws", splclient.NewMockAWSS3Client)
+
+	initFunc := getClientWrapper.GetS3ClientInitFuncPtr()
+
+	s3ClientMgr, err = getS3ClientMgr(client, &cr, &cr.Spec.AppFrameworkConfig, "appSrc1")
+	if err != nil {
+		t.Errorf("unable to get S3ClientMgr instance")
+	}
+
+	s3ClientMgr.initFn = initFunc
+
+	worker.waiter.Add(1)
+	go worker.Download(pplnPhase, *s3ClientMgr, &activeWorkers, "")
+	worker.waiter.Wait()
+	// we should return error here
+	if ok, _ := areAppsDownloadedSuccessfully(appDeployInfoList); ok {
+		t.Errorf("We should have returned error here since objectHash is empty in the worker")
+	}
+
+}
+
+func TestScheduleDownloads(t *testing.T) {
+	var ppln *AppInstallPipeline
+	appDeployContext := &enterpriseApi.AppDeploymentContext{}
+	ppln = nil
+	ppln = initAppInstallPipeline(appDeployContext)
+	ppln.availableDiskSpace = 100
+	pplnPhase := ppln.pplnPhases[enterpriseApi.PhaseDownload]
+
+	cr := enterpriseApi.Standalone{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "s1",
+			Namespace: "test",
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Standalone",
+		},
+		Spec: enterpriseApi.StandaloneSpec{
+			Replicas: 1,
+			AppFrameworkConfig: enterpriseApi.AppFrameworkSpec{
+				AppsRepoPollInterval:      60,
+				MaxConcurrentAppDownloads: 5,
+
+				VolList: []enterpriseApi.VolumeSpec{
+					{
+						Name:      "test_volume",
+						Endpoint:  "https://s3-eu-west-2.amazonaws.com",
+						Path:      "testbucket-rs-london",
+						SecretRef: "s3-secret",
+						Provider:  "aws",
+					},
+				},
+				AppSources: []enterpriseApi.AppSourceSpec{
+					{
+						Name:     "appSrc1",
+						Location: "adminAppsRepo",
+						AppSourceDefaultSpec: enterpriseApi.AppSourceDefaultSpec{
+							VolName: "test_volume",
+							Scope:   "local",
+						},
+					},
+					{
+						Name:     "appSrc2",
+						Location: "securityAppsRepo",
+						AppSourceDefaultSpec: enterpriseApi.AppSourceDefaultSpec{
+							VolName: "test_volume",
+							Scope:   "local",
+						},
+					},
+					{
+						Name:     "appSrc3",
+						Location: "authenticationAppsRepo",
+						AppSourceDefaultSpec: enterpriseApi.AppSourceDefaultSpec{
+							VolName: "test_volume",
+							Scope:   "local",
+						},
+					},
+				},
+			},
+		},
+	}
+	sts := &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "splunk-s1-standalone",
+			Namespace: "test",
+		},
+	}
+	sts.Spec.Replicas = new(int32)
+	*sts.Spec.Replicas = 1
```

Comment:
Do we need a test with multiple standalone replicas?  Or maybe this was part of another commit or I missed it.